api/app/registration/api.py

```diff
@@ -0,0 +1,296 @@
+from datetime import datetime
+import traceback
+from flask_restful import fields, marshal_with
+import flask_restful as restful
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.exc import SQLAlchemyError
+from app.utils import errors
+from app import LOGGER
+from app.registration.models import RegistrationSection
+from app.registration.models import RegistrationQuestion
+from app.registration.models import RegistrationForm
+from app.registration.models import Offer
+from app.registration.mixins import RegistrationFormMixin, RegistrationSectionMixin, RegistrationQuestionMixin
+from app.events.models import Event
+from app import db
+
+
+def registration_form_info(registration_form):
+    return {
+        'registration_form_id': registration_form.id,
+        'event_id': registration_form.event_id
+    }
+
+
+class RegistrationFormAPI(RegistrationFormMixin, restful.Resource):
+
+    registration_question_fields = {
+        'description': fields.String,
+        'type': fields.String,
+        'is_required': fields.Boolean,
+        'order': fields.Integer
+    }
+
+    registration_section_fields = {
+        'id': fields.Integer,
+        'name': fields.String,
+        'description': fields.String,
+        'order': fields.Integer,
+        'registration_questions': fields.List(fields.Nested(registration_question_fields))
+    }
+
+    registration_form_fields = {
+        'event_id': fields.Integer,
+        'registration_sections': fields.List(fields.Nested(registration_section_fields))
+    }
+
+    @marshal_with(registration_form_fields)
+    def get(self):
+        args = self.req_parser.parse_args()
+        event_id = args['event_id']
+        offer_id = args['offer_id']
+
+        try:
+            offer = db.session.query(Offer).filter(Offer.id == offer_id).first()
+            if offer and offer.expiry_date >= datetime.now():
+                registration_form = db.session.query(RegistrationForm).filter(
+                    RegistrationForm.event_id == event_id).first()
+
+                if not registration_form:
+                    return errors.REGISTRATION_FORM_NOT_FOUND
+
+                sections = db.session.query(RegistrationSection).filter(
```

Comment:
We need to filter the sections according to the flags on the offer. I see this wasn't included in the issue - sorry about that! The relevant text from the spec is:

> Note that the relevant controller that returns a registration form, should compare the show_for_* flags on each RegistrationSection against the relevant Offer. If all of the show_for_* flags are NULL, the RegistrationSection should be included in the response. If any of the show_for_* flags are not null, the RegistrationSection should only be included if the condition `(RegistrationSection.show_for_travel_award AND Offer.travel_award) OR (RegistrationSection.show_for_accomodation_award AND Offer.accomodation_award) OR (RegistrationSection.show_for_payment_required AND Offer.payment_required) is TRUE`.